hazelcast/src/test/java/com/hazelcast/quorum/list/AbstractListQuorumTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.quorum.list;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.config.ListConfig;
+import com.hazelcast.config.QuorumConfig;
+import com.hazelcast.core.IList;
+import com.hazelcast.quorum.PartitionedCluster;
+import com.hazelcast.quorum.QuorumType;
+import com.hazelcast.test.TestHazelcastInstanceFactory;
+
+import static com.hazelcast.quorum.PartitionedCluster.QUORUM_ID;
+import static com.hazelcast.test.HazelcastTestSupport.randomMapName;
+
+public abstract class AbstractListQuorumTest {
+
+    protected static final String LIST_NAME_PREFIX = "quorum";
+    protected static final String LIST_NAME = randomMapName(LIST_NAME_PREFIX);
+
+    protected static final Object lock = new Object();
+
+    protected static PartitionedCluster cluster;
```

Comment:
Is there a happens before relation missing here as well? I don't understand why the lock is needed; but if it is needed then multiple threads are probably calling the init and a different thread running the tests. I'm not sure if any outside happens before relation is provided, but probably simplest to make this field volatile.